src/NuGet.Clients/NuGet.VisualStudio.Common/OutputConsoleLogger.cs

```diff
@@ -167,13 +166,6 @@ private async Task<int> GetMSBuildVerbosityLevelAsync()
 
         public void ReportError(ILogMessage message)
         {
-            Run(() => ReportAsync(message));
-        }
-
-        private async Task ReportAsync(ILogMessage message)
-        {
-            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-
             var errorListEntry = new ErrorListTableEntry(message);
```

Comment:
> Is it ok not to change into main thread?

Usually it's not. In this particular case, it's ok because the assumption was incorrect. 
If you look at some other codepaths, you'd observe that the same method is already called on a background thread.

> By the way which one is this "outputConsoleLogger"? Is it below one?

Yes, the `Package Manager` output window is ours.

Reply:
I'd guess the bg thread works because we're just adding to that error collection, and VS probably marshals that to the UI thread anyway.

> By the way which one is this "outputConsoleLogger"? Is it below one?

Correct, but note that the code you pointed out is actually sending to the Error List.

Reply:
> I'd guess the bg thread works because we're just adding to that error collection, and VS probably marshals that to the UI thread anyway.

If you follow the stacks, you'll find that we're calling `ITableDataSink.AddEntries`. 

The description is: https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.shell.tablemanager.itabledatasink.addentries?view=visualstudiosdk-2019#Microsoft_VisualStudio_Shell_TableManager_ITableDataSink_AddEntries_System_Collections_Generic_IReadOnlyList_Microsoft_VisualStudio_Shell_TableManager_ITableEntry__System_Boolean_

Some component somewhere ensures that this data is shown, but that was not a consideration for me :) 
The data is not synchronously pushed to the UI thread in this call, so that's why it's safe to just call it from any thread.